async-nats/src/jetstream/message.rs

```diff
@@ -41,6 +41,19 @@ impl From<Message> for crate::Message {
 }
 
 impl Message {
+    /// Splits [Message] into [Acker] and [crate::Message].
+    /// This can help reduce memory footprint if [Message] can be dropped before acking,
+    /// for example when it's transformed into another structure and acked later
+    pub fn acker(self) -> (crate::Message, Acker) {
```

Comment:
Bike-shedding, maybe?

```suggestion
    pub fn split(self) -> (crate::Message, Acker) {
```

Tangentially, feel like we should consider renaming `jetstream::Message` to represent that it is (conceptually) a tuple of an acker and message, bit of a PITA if you have both message types in scope.